### ISSUE TITLE
Clicking Clear All > Finished or User Actions > Send Message doesn't do anything.

### ISSUE THREAD

**Issue by rpmindroid** [nicotine-plus/nicotine-plus#2617]
Nicotine+ version: latest build as of Aug 10
Operating System/Distribution: macOS 13.4.1

### Describe the bug

1. Some elements of the context menu are unresponsive. Once the element is clicked, the menu disappears but nothing happens. I noticed this for:
'Clear All' > 'Finished'
'Clear All' > 'Finished / Filtered'
'User Actions' > 'Send Message' (whether I'm in Search or in Downloads)

Others like 'Download', 'Browse Folders', 'File Properties' in Search, or 'Clear' in Downloads, work fine.

2. Sometimes after moving the cursor from 'Copy' to 'User Actions' (in Search) or to 'Clear All' (in Downloads), the 'Copy' menu entry stays greyed/activated and open (see screenshots - the cursor don't appear on it but they're on 'User Actions' resp. 'Clear All').

3. Also sometimes, I think after going into another window and coming back to N+, nothing you can click in the N+ window is responsive (e.g. menu bar buttons or context menus left open before moving to the other window). They 'grey up' when you hover over them, but you can't click them. I've noticed this only a couple of times, and I can't reproduce it, but I thought I'd mention it in case it's related to 2.

### Expected behavior

1. Clicking said buttons would trigger their functionalities.
2. 'Copy' entry would collapse and become white again as the cursor moves on to 'User Actions' resp. 'Clear All'.

### Steps to reproduce the bug
1. Go to the Search or Downloads tabs, right-click on some item in your search results resp. download queue, and try to click 'Clear All' > 'Finished', 'Clear All' > 'Finished / Filtered', or 'User Actions' > 'Send Message'.
2. Not sure how exactly, but I manage to trigger it randomly pretty often.

### Additional context

Could it be related to the cursor issue mentioned there?
https://github.com/nicotine-plus/nicotine-plus/issues/2298
https://github.com/nicotine-plus/nicotine-plus/issues/2587

The buttons for 'Clear Finished' and 'Clear All' > ... in the bottom-right corner of the window do work.
The backspace key doesn't seem to (if it's supposed to).

<img width="830" alt="Screenshot 2023-08-10 at 21 57 31" src="https://github.com/nicotine-plus/nicotine-plus/assets/93481982/f80c8b9e-fe40-41b7-8604-f97775d562f4">

<img width="830" alt="Screenshot 2023-08-10 at 22 16 27" src="https://github.com/nicotine-plus/nicotine-plus/assets/93481982/a3753b3c-2a49-4eb8-a670-c527f4690119">

**Comment by mathiascode** [comment]
I recall seeing some issue/discussion in the GTK repository about submenus not working properly on macOS. I'll look into it once I have a working macOS VM again.

**Comment by mathiascode** [comment]
https://gitlab.gnome.org/GNOME/gtk/-/issues/5305
https://gitlab.gnome.org/GNOME/gtk/-/issues/5905
https://gitlab.gnome.org/GNOME/gtk/-/issues/5913

**Comment by slook** [comment]
@rpmindroid Does the issue occur when using the cursor arrow keys on the keyboard to navigate the sub-menus, or not?

**Comment by rpmindroid** [comment]
> @rpmindroid Does the issue occur when using the cursor arrow keys on the keyboard to navigate the sub-menus, or not?

It doesn't actually. :)

- if I don't move the mouse cursor over the context menu after the right-click, and only use the arrow keys to navigate main- and submenus, everything works fine (clear all, send message).

- if I use the mouse cursor to hover over the main context menu and open a submenu, and then the arrow keys to navigate that submenu, everything works fine as well.

So it's really just the clicking inside of submenus that doesn't work.

Also the 'left' arrow key doesn't just collapse a submenu to go back into the main one, it makes the whole menu disappear - which is maybe not a bug but rather a design flaw? - but I can just use the mouse cursor inside the main menu to move on and make submenus collapse, so that's not really a problem.

- if I use both cursors over the main context menu, after a random while (ranging from 0.1 to several seconds), the arrow keys cursor becomes inactive.

So basically, using the mouse over the main menu and the keys inside the submenus solves every problem. :)

**Comment by mathiascode** [comment]
I've seen comments about similar issues in GTK apps being fixed in macOS 13.5. OP has removed their account, but perhaps @WesCurry (if you're on macOS Ventura) could confirm if this is the case?

**Comment by WesCurry** [comment]
> I've seen comments about similar issues in GTK apps being fixed in macOS 13.5. OP has removed their account, but perhaps @WesCurry (if you're on macOS Ventura) could confirm if this is the case? Could you also check if the window resize cursors behave properly (don't get stuck)?

I have similar issues to OP however I've found ways to work around it, so for example if I want to browse someones shares I just look up their user profile and then browsing works from there. Compared to not being able to right click on a file and then browse the shares from the search box.

A lot of my issues seem to be fixed in the latest Testing DMG though so I would encourage OP to try that.

There are a lot of minor issues with the interface but overall the latest build is working OK

**Comment by mathiascode** [comment]
> There are a lot of minor issues with the interface but overall the latest build is working OK

What are the remaining issues? If they are bugs in GTK, it would be nice to get them reported and fixed.
